sourcemash/frontend/static/js/views/browse_view.js

```diff
@@ -32,24 +33,33 @@ Sourcemash.Views.BrowseView = Backbone.View.extend({
                                     "Source": 'browse' });
   },
 
+  preserveURL: function(e){
+    e.preventDefault();
+    window.location.href = e.target.href;
+    window.location.replace('/#browse');
+  },
```

Comment:
1. Stops the href from HTML from going.
2. Extracts the href from the HTML and goes there.
3. Replaces the address bar with "/#browse"